Scripts/BuildPhases/LintAppLocalizedStringsUsage.swift

```diff
@@ -252,16 +246,14 @@ extension Xcodeproj {
             PBXSourcesBuildPhase.self,
             PBXBuildFile.self
         ]
-
+        
         init(from decoder: Decoder) throws {
-            // Try to decode each known types in turn, until we find one that succeeds decoding — by having the expected `isa` field value.
-            for objectType in Self.knownTypes {
-                if let object = try? objectType.init(from: decoder) as PBXObject {
-                    self.wrappedValue = object
-                    return
-                }
+            let untypedObject = try UnknownPBXObject(from: decoder)
+            if let objectType = Self.knownTypes.first(where: { $0.isa == untypedObject.isa }) {
+                self.wrappedValue = try objectType.init(from: decoder)
+            } else {
+                self.wrappedValue = untypedObject
             }
-            self.wrappedValue = try UnknownPBXObject(from: decoder) as PBXObject // Fallback
         }
```

Comment:
The new trick (instead of the `ISA<Self>` we had previously) is simply to call `init(from:)` on 2 types (`UnknownPBXObject` then the dynamic `objectType` type.

Each call to `init(from:)` at that level will each request a `KeyedDecodingContainer` on the same "dictionary" found at the same level in the plist (it's just that each type will be using `decoder.container(keyedBy:)` with different `CodingKeys` for each).

Which wouldn't be achievable if we implemented this `PBXObjectWrapper.init(from decoder: Decoder)` using the classical pattern of calling `decoder.container(keyedBy: ISAKey.self)` then `container.decode(objectType, forKey: ???)` to try to decode the rest of the fields (and is what made all this "decode object type based on one its field" a challenge to begin with)

---

Some benefits of this approach compared to the old one:
 - It should be more performant, as it decodes the `isa` field as a `String` (under the hood via `UnknownPBXObject(from: decoder)`, then immediately known which `objectType` to try to decode from there, as opposed to having to try to decode all the known types and let many fail until we find one that doesn't, like we did before
 - It also allows better error handling, because if we have a case of `isa = "PBXFileReference"` but for some reason (e.g. corrupted `pbxproj` file) the other fields were invalid and not matching an actual `PBXFileReference` making `PBXFileReference(from: decoder)` fail to decode, then:
    - The previous approach would just silently eat that error and fallback to a `UnknownPBXObject` instead — despite us knowing about what the `PBXFileReference` is supposed to be and wanting our script to handle them — and would not report the inconsistency and the project being corrupted
    - While with that new approach, if the `isa` tells us that it should be a `PBXFileReference` but we fail to decode it as such, we can `throw` a proper corresponding error instead of silently ignoring the inconsistency.

And obviously one last benefit is that it's much simpler… but it just took me some time to figure out that this would be the working solution (as it's not following the standard pattern for `init(from:)` implementations and is a bit unconventional)